### PR TITLE
Add several languages support for source code blocks

### PR DESCRIPTION
**Pull request by zetashift**
Zig and Erlang aren't working, but I have no idea why? Erlang is available in: https://github.com/Microsoft/vscode/blob/master/extensions/markdown-basics/syntaxes/markdown.tmLanguage.json for example.
@AdrieanKhisbe if you have time could you please look into why Erlang for example isn't working but Elixir is?

### CODE REVIEW

File: syntaxes/populate-syntax.js [hunk 1]
```diff
@@ -39,7 +39,12 @@ const programmingLanguages = [
     ['shell', 'sh', 'bash', 'zsh', 'bashrc', 'bash_profile'],
     ['cs', 'csharp', 'c#'],
     ['fs', 'fsharp', 'f#'],
-    ['dart']
+    ['dart'],
+    ['nim'],
+    ["elixir"],
+    ["erlang"],
+    ["ocaml"],
+    ["zig"],
```
Comment (AdrieanKhisbe) [inline syntaxes/populate-syntax.js]: please use single quote

File: syntaxes/populate-syntax.js [hunk 2]
```diff
@@ -39,7 +39,12 @@ const programmingLanguages = [
     ['shell', 'sh', 'bash', 'zsh', 'bashrc', 'bash_profile'],
     ['cs', 'csharp', 'c#'],
     ['fs', 'fsharp', 'f#'],
-    ['dart']
+    ['dart'],
+    ['nim'],
+    ["elixir"],
+    ["erlang"],
```
Comment (AdrieanKhisbe) [inline syntaxes/populate-syntax.js]: according to erlang grammar, here are the extension used
```suggestion
    ['erlang', 'erl', 'hrl'],
```

File: syntaxes/org.tmLanguage.json [hunk 3]
```diff
@@ -1309,6 +1309,171 @@
...
+                        }
+                    },
+                    "patterns": [
+                        {
+                            "begin": "(^|\\G)(\\s*)(.*)",
+                            "while": "(?i)(^|\\G)(?!\\s*#\\+END_SRC\\s*)",
+                            "contentName": "meta.embedded.block.zig",
+                            "patterns": [
+                                {
+                                    "include": "source.zig"
```
Comment (AdrieanKhisbe) [inline syntaxes/org.tmLanguage.json:1471]: about `zig`, seems to be correct according vscode zig extension grammar: https://github.com/ziglang/vscode-zig/blob/master/syntaxes/zig.tmLanguage.json#L4

To be sure, have you the extension installed?

File: syntaxes/org.tmLanguage.json [hunk 4]
```diff
@@ -1309,6 +1309,171 @@
...
+                                {
+                                    "include": "source.elixir"
+                                }
+                            ]
+                        }
+                    ]
+                },
+                {
+                    "name": "meta.block.source.erlang.org",
+                    "begin": "(?i)(?:^|\\G)(?:\\s*)(#\\+BEGIN_SRC)\\s+(erlang)\\b\\s*(.*)$",
```
Comment (AdrieanKhisbe) [inline syntaxes/org.tmLanguage.json]: about erlang, might be related to case issue. Did you use `erlang` or `Erlang`?
Regex implementation is case sensible contrary to markdown one:

 https://github.com/Microsoft/vscode/blob/master/extensions/markdown-basics/syntaxes/markdown.tmLanguage.json#L1719

**Review by AdrieanKhisbe — CHANGES_REQUESTED** [review]
Thanks @zetashift for the PR,

here are some guidelines and some tracks why that might not be working for `erlang` and `zig`

File: syntaxes/populate-syntax.js (hunk 1 above)
Comment (zetashift) [reply syntaxes/populate-syntax.js]: Whoops force of habit sorry.

**Comment by zetashift** [comment]
I did not have the necessary extensions installed. Everything is working now. I don't know anymore languages but if people here are lurking know of any, let me know!

**Comment by zetashift** [comment]
Not sure why the CI is failing: `AssertionError [ERR_ASSERTION]: '* Header\n' +`

Does this have to do with line endings and different OS's?

**Comment by AdrieanKhisbe** [comment]
> Not sure why the CI is failing: `AssertionError [ERR_ASSERTION]: '* Header\n' +`
> 
> Does this have to do with line endings and different OS's?

@zetashift Test suite is not working on macos on master neither: https://travis-ci.com/github/vscode-org-mode/vscode-org-mode/jobs/479782572,

CI might need to be updated

File: package-lock.json [hunk 5]
```diff
@@ -1,8 +1,5533 @@
 {
```
Comment (AdrieanKhisbe) [inline package-lock.json]: Lock should not be updated in the PR, only syntax and generating script should be

**Review by AdrieanKhisbe — CHANGES_REQUESTED** [review]
Except than the `package-lock` extra, seems good to me :)

File: package-lock.json (hunk 5 above)
Comment (zetashift) [reply package-lock.json]: I'm not even sure how that happen D: , my JS-packaging knowledge is limited and I use yarn too. I thought package-lock.json never changed suddenly too.
I just replaced package.json and package-lock.json with the ones from the repo.

**Comment by zetashift** [comment]
@AdrieanKhisbe did I correct the package-lock.json correctly?

**Comment by AdrieanKhisbe** [comment]
> @AdrieanKhisbe did I correct the package-lock.json correctly?

Yes indeed @zetashift , I haven't noticed, a bit overwhelm these days

**Review by AdrieanKhisbe — APPROVED** [review]
It's good to me.

(CI broken on macos, but unrelated to that pull request)

**Comment by zetashift** [comment]
Thanks for the review, I hope you get a break/rest soon!

**Pull request merged**